app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -165,6 +165,29 @@ class MainActivity : BottomSheetActivity(), ActionButtonActivity, HasAndroidInje
         val activeAccount = accountManager.activeAccount
             ?: return // will be redirected to LoginActivity by BaseActivity
 
+        // As early as possible register to be notified when the service saves a "couldn't upload" draft.
+        // This will be called when the app first comes up, which is desirable if a draft has been waiting awhile.
```

Comment:
The place this is called is actively Bad. I put it at the very top of MainActivity because it needs to go somewhere that could be activated "anywhere". However it DEFINITELY should go after the intent check, and it needs some kind of awareness of places it "shouldn't" activate— it shouldn't pop up over anything that it would "break" by interrupting (A compose pane?) and it definitely shouldn't activate if the drafts pane is already open. I need guidance on when to call this and how to tell if we're in a "non-interruptable" pane and/or the drafts pane (would having the drafts pane just call clearDraftsAlert on open be enough??).

Reply:
Also note: The observer callback DOES get called unconditionally on boot / re-entry into the program, which is what we want.